Задание 2. Создан README

gudkxv opened the pull request:
Всем доброго дня, посылаю PR с выполненным заданием 2: "Создание README".

gudkxv committed:
Задание 2. Создан README

vaganov-vadim commented on README.md:
````diff
@@ -0,0 +1,23 @@
+```
+> ### COMING SOON
+> Данный файл будет расширяться в процессе разработки проекта. :frog:
+
````
Привет! Давай тут добавим схемку взаимодействия компонентов с помощью PlantUML. Это довольно важно, чтобы понимать место сервиса в общей архитектуре.

gudkxv replied:
Здравствуйте, исправил.
Появился вопрос: 
Нечаянно добавил лишний коммит(https://github.com/gpb-it-factory/gudkov-telegram-bot/pull/2/commits/c7d46875e0487a27176b60266f7b7e2f4c7a9991), я же правильно понимаю, что я при мердже смогу выбрать коммиты, которые нужно вливать, или мне нужно этот коммит вручную отменить?

vaganov-vadim replied:
Я расписывал по настройкам, что у нас должен быть squash+merge, то есть все коммиты сольются в один. Если что-то критичное залилось, то лучше поискать, как удалить коммит из истории и проделать эту операцию :)

vaganov-vadim reviewed (changes requested):
Привет! В целом хорошо, но давай добавим PlantUML-схемку взаимодействия компонентов. И хорошо бы не забыть про пользователя - как он будет включён во всю эту схему.

gudkxv committed:
Add scheme.puml

reworked photo.png to scheme.puml so scheme will be updated when file is updated

gudkxv committed:
Update README.md

Added UML component scheme

vaganov-vadim commented on scheme.puml:
```diff
@@ -0,0 +1,6 @@
+@startuml
```
Ну что-то совсем простенько и не совсем корректно. У нас же будет ещё Backend. Предлагаю добавить хоть немного конкретики - какой куда запрос, кому что возвращается. Хотя бы по твоим представлениям. Сейчас схема совсем общая и понимания проекта не добавляет.

vaganov-vadim reviewed (changes requested):
Предлагаю дополнить схему, сейчас на ней маловато информации.

Commit:
Reworked README. Added Sequence_diagram.puml, rebuild component_diagram.puml

Commit:
Linked to fc47846.Forgot to -a when commiting

gudkxv commented:
Постарался расширить UML.
Такой вопрос: для отображения PlantUML использую их прокси, который сейчас указывает на файлы в ветке feature. После мерджа, путь к .puml изменится с future/* на trunk/*. 
Для проверки оставил пути как есть, но, если я правильно понимаю, перед мерждем нужно будет еще коммитнуть изменение пути к .puml файлам, чтобы отображение моделей не упало?

vaganov-vadim commented:
> Постарался расширить UML. Такой вопрос: для отображения PlantUML использую их прокси, который сейчас указывает на файлы в ветке feature. После мерджа, путь к .puml изменится с future/* на trunk/*. Для проверки оставил пути как есть, но, если я правильно понимаю, перед мерждем нужно будет еще коммитнуть изменение пути к .puml файлам, чтобы отображение моделей не упало?

Я, честно говоря, таким способом не пользовался, я просто делаю ```plantuml`` в markdown, без всяких ссылок, он рендерится уже движками на сайтах/в IDE. Но наверное логичнее сразу указывать trunk.

gudkxv committed:
Update README.md

Changed .puml files links from /feature to /trunk branch so they work after merge. Removed inactive <div> tags from file.

gudkxv commented:
Я так понимаю, что для отображения через Markdown нужно какой-то плагин устанавливать. Я попытался - вроде понимает, что перед ним PlantUML, но рендерить не хочет. Поэтому решил оставить через прокси, ссылку заменил.

vaganov-vadim commented:
> Я так понимаю, что для отображения через Markdown нужно какой-то плагин устанавливать. Я попытался - вроде понимает, что перед ним PlantUML, но рендерить не хочет. Поэтому решил оставить через прокси, ссылку заменил.

Ок. На будущее - markdown extensions.
